### PR TITLE
Story/382578

### PR DESCRIPTION
**Pull request by sxu1998**
https://tfs/DefaultCollection/Server%20Tools/_workitems/edit/382578

in addition to the story above, i also created a file that will create a server session that is run before each test suite, and a file that will invalidate a server session that's run after each test suite instead of copy pasting the same lines of code in all the files

### CODE REVIEW

**Comment by terrencejames-lf** [comment]
I feel like in general there should be a way to call invalidate from one place vs copying and pasting in every test

File: test/Attributes/AttributeKeys.test.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { repoId } from '../testHelper.js';
-import { ODataValueContextOfListOfAttribute,IRepositoryApiClient } from '../../src/index.js';
+import { ODataValueContextOfListOfAttribute, IRepositoryApiClient } from '../../src/index.js';
 import { createTestRepoApiClient } from '../BaseTest.js';
 
 describe('Attribute Key Integration Tests', () => {
   let _RepositoryApiClient: IRepositoryApiClient;
   _RepositoryApiClient = createTestRepoApiClient();
+  afterAll(async () => {
```
Comment (terrencejames-lf) [inline test/Attributes/AttributeKeys.test.ts:8]: I feel like in general there should be a way to call invalidate from one place vs copying and pasting in every test

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts:8]: do you mean like create another function in the baseTest.ts file that will invalidate the server session, and in each describe block, call that function?

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts:8]: i think there are a couple of properties in the jest.config.js files that might allow me to just create a server session b4 all test suites and then invalidate after all test suites 

https://jestjs.io/docs/configuration#globalsetup-string
https://jestjs.io/docs/configuration#globalsetup-string

Comment (terrencejames-lf) [reply test/Attributes/AttributeKeys.test.ts:8]: I'm saying like, it should invalidate after every test, but only come from one file

Comment (terrencejames-lf) [reply test/Attributes/AttributeKeys.test.ts:8]: or at least, after every suite

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts:8]: hmm not sure if that is possible in jest, i guess i will have to dig a little deeper to see how i can configure jest to run a certain file after each test suite (i.e. test.ts file)

Comment (terrencejames-lf) [reply test/Attributes/AttributeKeys.test.ts:8]: Could you not use global setup?

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts:8]: global setup executes the given file before all test suites, the global teardown is the property that executes the given file after all test suites

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts:8]: so i just tried out the global teardown property in the jest.config.js file, while adding this setup.ts file and did an npm run for all the get entries file 
image
image
image

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts:8]: ok so now after running multiple test suites, it appears that the console.log statement doesn't get executed until all the test suites are finished running (as opposed it executing after each test suite)

image

Comment (sxu1998) [reply test/Attributes/AttributeKeys.test.ts]: ok i figured it out, it is these 2 parameters in the jest.config.js file below that are run before and after each test suite, so in addition to adding a seperate file for the invalidate api call (i.e. teardown.ts), I also added a file for creating a server session before each test suite (i.e. setup.ts)
image

**Review by terrencejames-lf — APPROVED** [review]

**Pull request merged**